src/classes/utils/tree-utils.ts

```diff
@@ -12,4 +13,32 @@ export class TreeUtils {
     ]
     return nodeList
   }
+
+  static addBreadcrumb(root: Root | Tree): void {
```

Comment:
After data increase, the compressed tree data is 1.8MB(compressed) for 26,290 docs.